question25/main.js

```diff
@@ -0,0 +1,115 @@
+const nameInputArea = document.getElementById('name');
+const mailInputArea = document.getElementById('mail');
+const passwordInputArea = document.getElementById('password');
+
+
+textLinkToTerms.addEventListener('click',()=>{
+    modal.classList.remove('hidden');
+    mask.classList.remove('hidden');
+});
+modalClose.addEventListener('click',()=>{
+    modal.classList.add('hidden');
+    mask.classList.add('hidden');
+});
+mask.addEventListener('click',()=>{
+    modalClose.click();
+});
+
+const options = {
+  root: document.getElementById('js-modal'),
+  threshold: 1
+};
+
+const checkWhenIntersect = ([entry]) => {
+  if (entry.isIntersecting) {
+    checkbox.checked = true;
+    checkbox.disabled = false;
+    switchSubmitButton(checkAllValidity());
+  };
+};
+
+const observer = new IntersectionObserver(checkWhenIntersect, options);
+observer.observe(modalContainer.lastElementChild);
+
+
+
+const isValidStatus = {
+  name: false,
+  mail: false,
+  password: false,
+}
+
+const validationInfo = {
+  name: {
+    maxNameLength: 16,
+    minNameLength: 1,
+    validation: (value) => value.length > validationInfo.name.minNameLength && value.length < validationInfo.name.maxNameLength,
+    errorMessage: '※ユーザー名は1文字以上15文字以下にしてください。',
+  },
+  mail: {
+    //Reference: https://1-notes.com/javascript-determine-if-it-is-an-email-address/ 
+    validation : (value) => /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(value),
+    errorMessage: '※メールアドレスの形式になっていません。' 
+  },
+  password: {
+    validation : (value) => /(?=.*[a-zA-Z])(?=.*[0-9])[a-zA-z0-9]{8,}/.test(value),
+    errorMessage: '※8文字以上の大小の英数字を交ぜたものにしてください。'  
+  }
+}
+
+const setValidationEvents = (isValid, targetForm) => {
+  if (isValid) {
+    targetForm.classList.add('valid');
+    targetForm.classList.remove('invalid');
+    targetForm.nextElementSibling.textContent = ''; 
+    isValidStatus[targetForm.id] = true;
+  } else {
+    targetForm.classList.add('invalid');
+    targetForm.classList.remove('valid');
+    targetForm.nextElementSibling.textContent = validationInfo[targetForm.id].errorMessage;
+    isValidStatus[targetForm.id] = false; 
+  }
+}
+
+const validateInputValue = (e) => {
+  const targetForm = e.target;
+  const value = targetForm.value.trim();
+  const result = validationInfo[targetForm.id].validation(value);
+  setValidationEvents(result,targetForm);
+  renderRequiredFieldMessages(targetForm);  
+  switchSubmitButton(checkAllValidity());
+} 
+
+const checkAllValidity = () => {
+  return Object.values(isValidStatus).every((result) => result );
+}
+
+const switchSubmitButton = (isValid) => {
+  submitButton.disabled = isValid && checkbox.checked ? false : true;
+} 
+
+const renderRequiredFieldMessages = (targetForm) => {
+  targetForm.nextElementSibling.textContent = targetForm.value.trim() === "" ? "※入力必須項目です" : "";
+};
```

Comment:
I had overwritten and deleted it after validation.... My apologies.

```suggestion
const renderRequiredFieldMessages = (targetForm) => {
  if (targetForm.value.trim() === "") {
    targetForm.nextElementSibling.textContent = "※入力必須項目です";
  }
};
```

Reply:
Thank you for that ideas! I had not noticed it either...! fixed so please confirm! 
fixed commit : https://github.com/csakyo/homework/pull/39/commits/1fc5f94274825e348dffedc6033eedc07f972b51